QUARKUS-1079.md

```diff
@@ -0,0 +1,74 @@
+# QUARKUS-1079 - Hibernate Reactive as Tech Preview
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-1079
+
+For native mode similar test coverage as for JVM mode needs to be ensured.
+
+## Scope of the testing
+This feature added previews of two extensions: 
+ - Hibernate Reactive
+ - Hibernate Reactive with Panache
+
+Following test coverage should be implemented for Hibernate Reactive:
+- Usage of application.properties file for passing reactive-hibernate settings [examples]
+- Usage of import.sql [examples]
+- Smoke test(getting a working Mutiny session) [examples]
+- Retrieving Multi result from the DB [examples]
+- Retrieving Uni result(require a transformation) from the DB
+- Usage of DB-generated ids and ids generated with Java code(via ReactiveIdentifierGenerator) [examples]
+- Validation of values [examples, doesn't work as expected]
+- Attribute converters
+- Usage of CompletionStages alongside a Mutiny API
+- Usage of session opened through openSession() method as well as withSession() [examples]
+- Usage of transaction started through openSession() method as well as withSession()
+- Usage of session and transaction opened with the same method
+- Session methods: find[examples], persist[examples], remove, merge, flush[examples]
+- Queries: regular[examples], native, named, JPA criteria
+- Query methods: setParameter[examples], setMaxResults, getFirstResult, getSingleResult[examples], getResultList[examples], executeUpdate
+- Eager[examples] and lazy fetches
+- Field-level lazy fetching(Hibernate team "doesn't encourage" using this feature, though)
+- Stateless sessions and their methods
+ 
+Following test coverage should be implemented for Hibernate Reactive with Panache:
+- Active record pattern
+- Repository pattern
+- list and stream methods
+- find method
+- sorting 
+- Named queries
+- DTO projection
+- Transactions
+- Id generation
+
+Ideally, all of aforementioned tests should run for PostgreSQL[examples], MySQL and DB2 connections.
+
+### Impact on testsuites and testing automation:
+ - Ensure this coverage works both in JVM and NATIVE mode
+ - Some smoke tests were created in "examples" section of quarkus-test-framework, more advanced should be put into quarkus-test-suite
```

Comment:
Test coverage should be mainly in https://github.com/quarkus-qe/quarkus-test-suite/

Examples directory should contain just one sample, it's not the main place for test coverage.

Reply:
Changed this section